.github/buildomat/jobs/tuf-repo.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+#:
+#: name = "helios / build TUF repo"
+#: variety = "basic"
+#: target = "helios-latest"
+#: output_rules = [
+#:	"=/work/manifest.toml",
+#:	"=/work/repo.zip.part*",
+#: ]
+#:
+#: [dependencies.ci-tools]
+#: job = "helios / CI tools"
+#:
+#: [dependencies.package]
+#: job = "helios / package"
+#:
+#: [dependencies.host]
+#: job = "helios / build OS image"
+#:
+#: [dependencies.trampoline]
+#: job = "helios / build trampoline OS image"
+#:
+
+set -o errexit
+set -o pipefail
+set -o xtrace
+
+COMMIT=$(git rev-parse HEAD)
+VERSION="1.0.0-alpha+git${COMMIT:0:11}"
+
+ptime -m gunzip < /input/ci-tools/work/tufaceous.gz > /work/tufaceous
+chmod a+x /work/tufaceous
+
+# Generate a throwaway repository key.
+python3 -c 'import secrets; open("/work/key.txt", "w").write("ed25519:%s\n" % secrets.token_hex(32))'
+read -r TUFACEOUS_KEY </work/key.txt
+export TUFACEOUS_KEY
+
+cat >/work/manifest.toml <<EOF
+system_version = "$VERSION"
+
+[artifact.control_plane]
+name = "control-plane"
+version = "$VERSION"
+[artifact.control_plane.source]
+kind = "composite-control-plane"
+EOF
+
+# switch-asic needs to be named "switch", skip here.
+# switch-softnpu does not go in TUF repos.
+for zone in $(ls /input/package/work/zones/* | grep -v switch); do
```

Comment:
A couple things came up today doing this manually to try mupdating the dogfood rack:

1. Some of the zones produced by `omicron-package` are only intermediates that are themselves then included in other zones; e.g., [dendrite](https://github.com/oxidecomputer/omicron/blob/857a9b95a83e7c833822fc29e3b89c4d1be41386/package-manifest.toml#L305) is marked as `intermediate_only` and is then a part of [the switch zone](https://github.com/oxidecomputer/omicron/blob/857a9b95a83e7c833822fc29e3b89c4d1be41386/package-manifest.toml#L360). Adding the intermediates here is relatively harmless, I think, other than making the tuf repo larger? But it would be good to prune them if we could.
2. For zones with multiple words (e.g., `crucible-pantry`), we may have some mixed hyphens/underscores? E.g., on my build VM, I see 

```
/work/zones/crucible-pantry.tar.gz
/work/zones/external-dns.tar.gz
/work/zones/internal-dns.tar.gz
/work/zones/mg-ddm.tar.gz
/work/zones/omicron-nexus.tar.gz
/work/zones/oximeter-collector.tar.gz
/work/zones/propolis-server.tar.gz
```

but on a running system under `/opt/oxide` I see

```
/opt/oxide/clickhouse.tar.gz
/opt/oxide/crucible_pantry.tar.gz
/opt/oxide/external_dns.tar.gz
/opt/oxide/oximeter.tar.gz
/opt/oxide/propolis-server.tar.gz
/opt/oxide/cockroachdb.tar.gz
/opt/oxide/internal_dns.tar.gz
/opt/oxide/nexus.tar.gz
/opt/oxide/ntp.tar.gz
/opt/oxide/crucible.tar.gz
/opt/oxide/mg-ddm.tar.gz
```

where some but not all of the hyphens have been replaced by underscores. I'm not sure what's doing that replacement today, or if I'm just confused?

Reply:
that is... strange. let me stare at package-manifest.toml and some other parts of the code

Reply:
OK, so dendrite wasn't ending up in the repo because the package job uses a [hardcoded list of zones](https://github.com/oxidecomputer/omicron/blob/15ae3e942e6eb90fbe6302e71f2367d542648ecb/.github/buildomat/jobs/package.sh#L72-L86).

The hyphens/underscores thing is because the zone names built in out/ and installed in /opt/oxide are inconsistent, e.g. the _package name_ is internal-dns but the _service name_ is internal_dns. Fortunately `omicron-package unpack` does all this logic for us and is trivial enough to get running in this job that we'll just use it.